drivers/nrf_radio_802154/platform/random/nrf_802154_random_zephyr.c

```diff
@@ -54,5 +89,5 @@ void nrf_802154_random_deinit(void)
 
 uint32_t nrf_802154_random_get(void)
 {
-    return sys_rand32_get();
+    return (uint32_t)next();
```

Comment:
Perhaps it's a dumb question, but wasn't it possible to use `entropy_get_entropy_isr`? I'm asking it because the commit message mentions only the problem of blocking on a semaphore.

Reply:
Not dumb at all, 
I haven't checked to be honest, I guess it should work but:
1. It's an entropy source, not RNG, I'm not convinced to use it as an RNG directly (which we do in Zephyr btw...),
2. I guess software RNG should be faster than HW generator (at least looking at https://github.com/zephyrproject-rtos/zephyr/blob/2f323e65b24283b75df2cd7d6e943340edd30f22/drivers/entropy/entropy_nrf5.c#L27, it takes it's time)

Reply:
I've rephrased the commit message slightly, hope it's more clear now.